### PR TITLE
Detect source audio format from ffmpeg log output

### PR DESCRIPTION
**Pull request by marcelveldt**
## Problem

Some providers don't supply (accurate) audio format details when creating
`StreamDetails`: bit rate is often missing for lossy codecs, sample rate / bit
depth fall back to the 44.1/16 defaults, and duration is occasionally zero or
absent (e.g. some podcast RSS feeds).

FFmpeg already probes all of this when it opens the input — we just weren't
parsing it. The stderr parser at `helpers/ffmpeg.py` previously only extracted
the codec; this PR extends it to also read sample rate, bit depth and bit rate
from the `Stream # … Audio:` line, plus the source duration from the
`Duration:` line, and mirrors those onto the streamdetails so downstream
consumers see the real source format.

## Changes

- Add `parse_ffmpeg_stream_info` / `parse_ffmpeg_duration` helpers in
  `helpers/ffmpeg.py`.
- Extract sample rate, bit depth (incl. the `s32 (24 bit)` FLAC case),
  bit rate and duration from ffmpeg's stderr output.
- Update `input_format` in place from the parsed values; `streamdetails`
  already shares that object so providers get the corrected values for free.
- Fill `streamdetails.duration` from ffmpeg when the provider didn't set one.
- Unit tests covering MP3, AAC, FLAC (16-bit and 24-bit), PCM, ALAC, Opus
  and edge cases (N/A duration, video stream lines).

### CODE REVIEW

File: music_assistant/controllers/streams/audio.py [hunk 1]
```diff
@@ -513,8 +513,16 @@ def _on_inband_metadata(metadata: dict[str, str]) -> None:
...
                     # for encoding the audio.
+                    # Note: ffmpeg_proc.input_format is the same object as
+                    # streamdetails.audio_format, so sample_rate / bit_depth / bit_rate
+                    # parsed from the ffmpeg log already live on streamdetails too.
                     first_chunk_received = True
                     streamdetails.audio_format.codec_type = ffmpeg_proc.input_format.codec_type
+                    # Some providers omit (or report 0 for) the item duration; ffmpeg can
+                    # usually probe it from the source. Only apply when missing so we
+                    # don't clobber an accurate provider value with a rounded one.
+                    if ffmpeg_proc.parsed_duration and not streamdetails.duration:
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/controllers/streams/audio.py]: good catch, fixed in f016d3cf1

[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/controllers/streams/audio.py]: duplicate of the one above, already fixed in f016d3cf1

File: music_assistant/helpers/ffmpeg.py [hunk 2]
```diff
@@ -209,6 +276,79 @@ async def _feed_stdin(self) -> None:
             if not generator_exhausted:
                 await close_async_generator(self.audio_input)
 
+    def _apply_input_stream_info(self, info: FFMpegStreamInfo) -> None:
+        """Mirror values from a parsed ffmpeg input stream line onto self.input_format."""
+        # content_type is the container format; only fill it in if the provider didn't
+        # specify one. codec_type is always set to what ffmpeg actually detected.
+        if self.input_format.content_type == ContentType.UNKNOWN:
+            self.input_format.content_type = info.codec
+        self.input_format.codec_type = info.codec
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/helpers/ffmpeg.py]: good catch — fixed in f68a155eb. also kept sample rate / bit depth / bit rate applying independently of the codec.

**Comment by OzGav** [comment]
This is great for the Audio Pipeline being able to show the truth of what is happening. Does this override manual settings like https://github.com/music-assistant/server/blob/3897b5135d3c65a43a7f4f97a7b3bdefa434bc0e/music_assistant/providers/motherearthradio/constants.py#L26
